modules/usage/module.php

```diff
@@ -415,23 +441,51 @@ private function add_to_global( $doc_name, $doc_usage ) {
 				];
 			}
 
-			$global_element_ref = &$global_usage[ $doc_name ][ $element_type ];
-			$global_element_ref['count'] += $element_data['count'];
+			$global_usage_ref = &$global_usage[ $doc_name ][ $element_type ];
+			$global_usage_ref['count'] += $element_usage_data['count'];
 
-			if ( empty( $element_data['controls'] ) ) {
+			if ( empty( $element_usage_data['controls'] ) ) {
 				continue;
 			}
 
-			foreach ( $element_data['controls'] as $tab => $sections ) {
+			foreach ( $element_usage_data['controls'] as $tab => $sections ) {
 				foreach ( $sections as $section => $controls ) {
 					foreach ( $controls as $control => $count ) {
-						$this->increase_controls_count( $global_element_ref, $tab, $section, $control, $count );
+						$this->increase_elements_controls_count( $global_usage_ref, $tab, $section, $control, $count );
 					}
 				}
 			}
 		}
 
-		update_option( self::OPTION_NAME, $global_usage, false );
+		update_option( self::ELEMENTS_OPTION_NAME, $global_usage, false );
+	}
+
+	private function remove_document_settings_from_global( $document ) {
+		$prev_usage = $document->get_meta( self::DOCUMENT_META_KEY );
+
+		if ( empty( $prev_usage ) ) {
+			return;
+		}
+
+		$doc_name = $document->get_name();
+
+		$global_usage = get_option( self::DOCUMENT_OPTION_NAME, [] );
+
+		foreach ( $prev_usage as $control_name => $control_count ) {
+			if ( ! empty( $global_usage[ $doc_name ][ $control_name ] ) ) {
+				$global_usage[ $doc_name ][ $control_name ] -= $control_count;
+
+				if ( 0 === $global_usage[ $doc_name ][ $control_name ] ) {
+					unset( $global_usage[ $doc_name ][ $control_name ] );
+				}
+			}
+		}
+
+		$global_usage = Utils::remove_empty_array_recursive( $global_usage );
```

Comment:
Maybe add a method to the Collection class that removes empty recursive, or `filter_recursive`

Reply:
Handled since didn't use collection.

Reply:
Handled.